keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,244 @@
+            # to be cropped from the original image
+            [0],  # box_indices: maps boxes to images along batch axis
+            # [0] since there is only one image
+            self.target_size,  # output size
+        )
+
+        return tf.squeeze(augmented_image, axis=0)
+
+    def _resize(self, image):
+        outputs = tf.keras.preprocessing.image.smart_resize(image, self.target_size)
+        # smart_resize will always output float32, so we need to re-cast.
+        return tf.cast(outputs, self.compute_dtype)
+
+    def _check_class_arguments(
+        self, target_size, crop_area_factor, aspect_ratio_factor
+    ):
+        if not isinstance(target_size, tuple):
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if len(target_size) != 2:
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if not isinstance(target_size[0], int):
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if not isinstance(target_size[1], int):
+            raise ValueError(
+                "`target_size` must be tuple of two integers."
+                f"Received target_size={target_size}"
+            )
+
+        if isinstance(aspect_ratio_factor, core.NormalFactorSampler):
+            raise ValueError(
+                "`aspect_ratio_factor` is an instance of NormalFactorSampler."
+                "Expected `aspect_ratio_factor` to be a tuple, ConstantFactorSampler or UniformFactorSampler."
+            )
+
+        if isinstance(crop_area_factor, core.NormalFactorSampler):
+            raise ValueError(
+                "`crop_area_factor` is an instance of NormalFactorSampler."
+                "Expected `crop_area_factor` to be a tuple, ConstantFactorSampler or UniformFactorSampler."
+            )
+
+    def get_config(self):
+        if isinstance(self.crop_area_factor, core.ConstantFactorSampler):
+            area_lower = self.crop_area_factor.value
+            area_upper = self.crop_area_factor.value
+        else:
+            area_lower = self.crop_area_factor.lower
+            area_upper = self.crop_area_factor.upper
+
+        if isinstance(self.aspect_ratio_factor, core.ConstantFactorSampler):
+            aspect_lower = self.aspect_ratio_factor.value
+            aspect_upper = self.aspect_ratio_factor.value
+        else:
+            aspect_lower = self.aspect_ratio_factor.lower
+            aspect_upper = self.aspect_ratio_factor.upper
+
+        config = super().get_config()
+        config.update(
+            {
+                "target_size": self.target_size,
+                "crop_area_factor": (area_lower, area_upper),
```

Comment:
you can literally pass self.crop_area_factor and self.aspect_area_factor in place of these.

Factors are already serializable.